stdpopsim/genomes.py

```diff
@@ -223,6 +223,12 @@ class Contig:
     genomic_element_types = attr.ib(factory=list)
     mutation_types = attr.ib(factory=list)
 
+    @recombination_map.validator
+    def _check_recombination_map_is_discrete(self, _attribute, _value):
+        discrete_position = np.array(self.recombination_map.position, dtype=int)
+        if any(discrete_position != self.recombination_map.position):
+            raise ValueError("recombination map must be discrete")
```

Comment:
I'm not sure if we should have this check, actually - regardless of whether the positions in the recombination map are integers, the resulting tree sequence will still have breakpoints at integer values. I guess that a recombination map with noniteger values might be in error, but the problem isn't because it's not discrete. (And, it'd be easy to do, e.g. if there was a breakpoint at `ts.sequence_length / 2`...)

Reply:
I agree - the breakpoints in the recombination map will have no effect on the output coordinates.